src/Common/src/CoreLib/System/IO/PathInternal.cs

```diff
@@ -10,13 +10,30 @@ internal static partial class PathInternal
         /// <summary>
         /// Returns true if the path ends in a directory separator.
         /// </summary>
-        internal static bool EndsInDirectorySeparator(ReadOnlySpan<char> path) => path.Length > 0 && IsDirectorySeparator(path[path.Length - 1]);
+        internal static bool EndsInDirectorySeparator(ReadOnlySpan<char> path)
+            => path.Length > 0 && IsDirectorySeparator(path[path.Length - 1]);
 
         /// <summary>
         /// Returns true if the path starts in a directory separator.
         /// </summary>
         internal static bool StartsWithDirectorySeparator(ReadOnlySpan<char> path) => path.Length > 0 && IsDirectorySeparator(path[0]);
 
+        internal static string EnsureTrailingSeparator(string path)
```

Comment:
These string overloads, which end up calling RoS methods will result in redundant casts from string to RoS. It would be better if we cast from string to RoS once at the highest level and just have all RoS internal helpers.

Reply:
This one modifies the path, it can't be a ROS. Eventually we're going to try and use buffers internally, but for some of these overloads we actually return the modified string to the user.  As such we will be stuck with some string helpers to avoid the unnecessary string allocations when the path isn't actually modified.

Reply:
Ah yes, good point. Forgot the "readonly" part.